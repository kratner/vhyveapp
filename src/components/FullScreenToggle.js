import React from "react";
import { AiOutlineFullscreenExit, AiOutlineFullscreen } from "react-icons/ai";
import styled from "styled-components";

const FullScreenIconInactive = styled(AiOutlineFullscreen)`
  fill: #fff;
  height: 2em;
  width: 2em;
  &:hover {
    fill: #fdde6b;
  }
`;
const FullScreenIconActive = styled(AiOutlineFullscreenExit)`
  fill: #fff;
  height: 2em;
  width: 2em;
  &:hover {
    fill: #fdde6b;
  }
`;
const FullScreenIconContainer = styled.div`
  cursor: pointer;
`;

class FullScreenToggle extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      fullScreen: false
    };
  }
  handleClick() {
    debugger;
  }
  render() {
    return (
      <FullScreenIconContainer
        id={this.props.id}
        onClick={e => {
          this.setState({
            fullScreen: !this.state.fullScreen
          });
        }}
        title={this.state.fullScreen ? "View Full Screen" : "Exit Full Screen"}
      >
        {//Check if message failed
        this.state.fullScreen ? (
          <FullScreenIconActive />
        ) : (
          <FullScreenIconInactive />
        )}
      </FullScreenIconContainer>
    );
  }
}

export default FullScreenToggle;
