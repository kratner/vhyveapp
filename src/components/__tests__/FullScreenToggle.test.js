import React from "react";
import { shallow } from "enzyme";
import FullScreenToggle from "../FullScreenToggle";

describe("FullScreenToggle", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<FullScreenToggle />);
    expect(wrapper).toMatchSnapshot();
  });
});
