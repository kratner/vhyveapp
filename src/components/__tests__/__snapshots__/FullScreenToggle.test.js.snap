// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenToggle renders correctly 1`] = `ShallowWrapper {}`;
